### PR TITLE
Ask user for tab to start

### LINKED ISSUE #1723: Make user choose what tab he wants to start on and make it default to news
**Wanna have the bug fixed quickly?**
Visit Issue hunt...
[Issue hunt](https://issuehunt.io/r/FAForever/downlords-faf-client)

### PR DESCRIPTION
**Pull request by 1-alex98**
User is asked what tab he wants to start in instead of taking last tab as starting point.
Fixes #1723

start_tab

### CODE REVIEW

**Comment by BlackYps** [comment]
The new comboboxes should have the same style as the other comboboxes in the client. e.g. the Database box right below in the settings

**Comment by 1-alex98** [comment]
already changed that, just the gif is old

**Comment by BlackYps** [comment]
A test is failing:
```
com.faforever.client.main.StartTabChooseControllerTest > unnecessary Mockito stubbings FAILED
    org.mockito.exceptions.misusing.UnnecessaryStubbingException
```

**Comment by 1-alex98** [comment]
yes gotta fix that

**Comment by anihilnine** [comment]
this is great. it would be best if we didnt launch it until the newsfeed is up

File: src/main/resources/i18n/messages.properties [hunk 1]
```diff
@@ -523,7 +523,7 @@ settings.giveFeedback=Open
 settings.windowTitle=Settings
 settings.general=General
 settings.language=Language
-settings.general.rememberLastTab=Remember my last opened tab
+settings.general.startTab=Select the tab to starts at
```
Comment (BlackYps) [inline src/main/resources/i18n/messages.properties:526]: Typo.
should be "Select the tab to start at"

File: src/main/resources/i18n/messages.properties [hunk 2]
```diff
@@ -759,3 +759,9 @@ replay.ignoreMapNotFound=Ignore and start replay
 replay.abortAfterMapNotFound=Abort and show more info
 replay.mapDownloadFailed=Map download failed for this replay
 replay.mapDownloadFailed.wannaContinue=We could not find the map mentioned in the replay file. However the game might still know where to find it. Consider this as a warning, that this replay might be broken.
+startTab.selectTab=Select the tab you want to starts at:
```
Comment (BlackYps) [inline src/main/resources/i18n/messages.properties]: See above

**Review by BlackYps — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/client/main/event/NavigateEvent.java [hunk 3]
```diff
@@ -1,7 +1,9 @@
 package com.faforever.client.main.event;
 
+import lombok.EqualsAndHashCode;
 import lombok.Getter;
 
+@EqualsAndHashCode
 public class NavigateEvent {
   @Getter
   private final NavigationItem item;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/main/event/NavigateEvent.java]: I'd suggest using `@Value` and `@AllArgsConstructor`.
Then remove `@EqualsAndHashCode`, remove `@Getter private final` and remove the constructor

File: src/main/java/com/faforever/client/main/event/NavigationItem.java [hunk 4]
```diff
@@ -1,16 +1,20 @@
 package com.faforever.client.main.event;
 
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
 import java.util.HashMap;
 
+@RequiredArgsConstructor
 public enum NavigationItem {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/main/event/NavigationItem.java]: This class also looks like a candidate for `@Value` (see above)

File: src/main/java/com/faforever/client/notification/ImmediateNotification.java [hunk 5]
```diff
@@ -1,33 +1,34 @@
...
+import lombok.RequiredArgsConstructor;
+
 import java.util.List;
 
 /**
  * A notification that requires the user's immediate attention. It is displayed until the user performs a suggested
  * action or dismisses it. The notification consists of a title, a text, an optional image and zero or more actions.
  */
+@RequiredArgsConstructor
 public class ImmediateNotification {
```
Comment (Brutus5000) [inline src/main/java/com/faforever/client/notification/ImmediateNotification.java:17]: This class also looks like a candidate for @Value (see above)

File: src/main/java/com/faforever/client/main/event/NavigationItem.java (hunk 4 above)
Comment (1-alex98) [reply src/main/java/com/faforever/client/main/event/NavigationItem.java]: Not possible on enums

**Pull request merged**